src/site/antora/modules/ROOT/pages/manual/configuration.adoc

```diff
@@ -14,1856 +14,1019 @@
-<Configuration status="debug" name="XMLConfigTest" packages="org.apache.logging.log4j.test">
-  <Properties>
-    <Property name="filename">target/test.log</Property>
-  </Properties>
-  <ThresholdFilter level="trace"/>
-
-  <Appenders>
-    <Console name="STDOUT">
-      <PatternLayout pattern="%m MDC%X%n"/>
-    </Console>
-    <Console name="FLOW">
-      <!-- this pattern outputs class name and line number -->
-      <PatternLayout pattern="%C{1}.%M %m %ex%n"/>
-      <filters>
-        <MarkerFilter marker="FLOW" onMatch="ACCEPT" onMismatch="NEUTRAL"/>
-        <MarkerFilter marker="EXCEPTION" onMatch="ACCEPT" onMismatch="DENY"/>
-      </filters>
-    </Console>
-    <File name="File" fileName="${filename}">
-      <PatternLayout>
-        <pattern>%d %p %C{1.} [%t] %m%n</pattern>
-      </PatternLayout>
-    </File>
-  </Appenders>
-
-  <Loggers>
-    <Logger name="org.apache.logging.log4j.test1" level="debug" additivity="false">
-      <ThreadContextMapFilter>
-        <KeyValuePair key="test" value="123"/>
-      </ThreadContextMapFilter>
-      <AppenderRef ref="STDOUT"/>
-    </Logger>
-
-    <Logger name="org.apache.logging.log4j.test2" level="debug" additivity="false">
-      <Property name="user">${sys:user.name}</Property>
-      <AppenderRef ref="File">
-        <ThreadContextMapFilter>
-          <KeyValuePair key="test" value="123"/>
-        </ThreadContextMapFilter>
-      </AppenderRef>
-      <AppenderRef ref="STDOUT" level="error"/>
-    </Logger>
-
-    <Root level="trace">
-      <AppenderRef ref="STDOUT"/>
-    </Root>
-  </Loggers>
-
-</Configuration>
-----
-
-[#PropertySubstitution]
-== Property Substitution
-
-Log4j 2 supports the ability to specify tokens in the configuration as
-references to properties defined elsewhere. Some of these properties
-will be resolved when the configuration file is interpreted while others
-may be passed to components where they will be evaluated at runtime. To
-accomplish this, Log4j uses variations of
-https://commons.apache.org/proper/commons-lang/[Apache Commons Lang]'s
-link:../javadoc/log4j-core/org/apache/logging/log4j/core/lookup/StrSubstitutor.html[`StrSubstitutor`]
-and
-link:../javadoc/log4j-core/org/apache/logging/log4j/core/lookup/StrLookup.html[`StrLookup`]
-classes. In a manner similar to Ant or Maven, this allows variables
-declared as `$\{name}` to be resolved using properties declared in the
-configuration itself. For example, the following example shows the
-filename for the rolling file appender being declared as a property.
+include::example$configuration/arbiters.properties[lines=17..-1]
+----
+====
+
+<1> If the Java system property `env` has a value of `dev`, a pattern layout will be used.
+<2> If the Java system property `env` has a value of `prod`, a JSON template layout will be used.
 
+The above example has a problem: if the Java system property `env` has a value different from `dev` or `prod`, the appender will have no layout.
```

Comment:
I don't think this deserves repeating a fully blown example a 2nd time. I'd rather only share `arbiters-select.xml` and explain the necessity of `DefaultArbiter`.

Reply:
Fixed in be54ca2aefd708b8159548277d99e12aea948980.